src/Core/Notifications/INotificationService.cs

```diff
@@ -18,10 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.Linq;
+using SonarLint.VisualStudio.Core.InfoBar;
+
 namespace SonarLint.VisualStudio.Core.Notifications
 {
-    public interface INotificationService
+    public interface INotificationService : IDisposable
     {
         void ShowNotification(INotification notification);
     }
+    
+    public sealed class NotificationService : INotificationService
+    {
+        /// <summary>
+        /// Taken from "ToolWindowGuids80.ErrorList"
+        /// </summary>
+        internal static readonly Guid ErrorListToolWindowGuid = new Guid("D78612C7-9962-4B83-95D9-268046DAD23A");
+
+        private readonly IInfoBarManager infoBarManager;
+        private readonly IThreadHandling threadHandling;
+
+        private Tuple<IInfoBar, INotification> activeNotification;
+
+        public NotificationService(IInfoBarManager infoBarManager, IThreadHandling threadHandling)
```

Comment:
Good question, it means the implementation would have to move to Integration (or we will need a MEF wrapper in Integration). I've been delaying the decision.. :) It doesn't make sense to make to move all this generic code into Integration just for MEF, but we have to MEF-import it because `InfoBarManager` needs VS stuff. Perhaps introducing MEF into Core would simplify things.